### PR TITLE
Refactor BatchID in its own module

### PR DESCRIPTION
**Pull request by fleupold**
Part of #842 

Pricegraph requires to be initialized with a BatchID. We already have an abstraction for this type in the scheduler. This PR moves this abstraction in the model module for reuse.

### Test Plan

CI

### CODE REVIEW

File: core/src/models/order.rs [hunk 1]
```diff
@@ -40,6 +41,29 @@ impl Order {
             self.remaining_sell_amount,
         )
     }
+
+    pub fn to_element(&self, balance: primitive_types::U256) -> Element {
+        // Some conversions are needed because the primitive types crate is on different versions in
```
Comment (nlordell) [inline core/src/models/order.rs:46]: I would say we should just pin the price graph version to be the same as core. Then we don’t need to do these conversions. It’s not really used anyway.

**Review by nlordell — APPROVED** [review]
Nice! My only comment is that we should make the `pricegraph` and `core` crates use the same `primitive-types` version.

Also, I’m a bit surprised the price graph needs a batch ID, but maybe I’m forgetting something.

**Comment by fleupold** [comment]
Sorry I included some unrelated code earlier. Updated the PR now

**Comment by fleupold** [comment]
> Also, I’m a bit surprised the price graph needs a batch ID, but maybe I’m forgetting something.

You are right, it's the OrderbookReading that needs it.

**Review by nlordell — APPROVED** [review]
Looks awesome now!

File: core/src/models/order.rs (hunk 1 above)
Comment (fleupold) [reply core/src/models/order.rs]: This code is actually meant to be in the next PR. Cleaned up the mess and repushed the change.

**Review by e00E — APPROVED** [review]

**Pull request merged**